### PR TITLE
refactor(i18n): i18n utils tools

### PR DESCRIPTION
**Pull request by EYHN**

### CODE REVIEW

**Comment by EYHN** [comment]
* **#7251** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7251?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @EYHN and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by JimmFly — APPROVED** [review]

File: packages/frontend/i18n/src/utils/time.ts [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+
+  if (relativeOption) {
+    // show relative
+
+    const formatter = new Intl.RelativeTimeFormat(i18n.language, {
+      style: 'narrow',
+      numeric: relativeOption.yesterdayAndTomorrow ? 'auto' : 'always',
+    });
+
+    const timeUnitProcessor = {
```
Comment (pengx17) [inline packages/frontend/i18n/src/utils/time.ts:113]: I think dayjs already provides relative time with locale support https://day.js.org/docs/en/plugin/relative-time
Do we need to do it on our own?

File: packages/frontend/i18n/src/i18n.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    } else {
+      return (self as any)[key as string] as any;
+    }
+  },
+});
+
+/**
+ * I18n['com.affine.xxx']({ arg1: 'hello' }) -> '中文 hello'
+ */
+export const I18n = I18nProxy as I18nFuncs & typeof I18nMethod;
```
Comment (pengx17) [inline packages/frontend/i18n/src/i18n.ts:53]: I am wondering function if localized strings with I18n will get updated when changing language in React

File: packages/frontend/i18n/src/utils/time.ts (hunk 1 above)
Comment (EYHN) [reply packages/frontend/i18n/src/utils/time.ts:113]: dayjs needs to load additional language files to implement i18n, using browser intl api can reduce the bundle size

File: packages/frontend/i18n/src/i18n.ts (hunk 2 above)
Comment (EYHN) [reply packages/frontend/i18n/src/i18n.ts:53]: The i18n here itself will change the language as react does, but the places where the i18n is used need to be re-executed.

Comment (pengx17) [reply packages/frontend/i18n/src/i18n.ts:53]: Then I guess using react hooks or <Trans /> component is still the recommended way of doing i18n in affine code, at least in react component?

Comment (EYHN) [reply packages/frontend/i18n/src/i18n.ts:53]: Yes

**Review by pengx17 — APPROVED** [review]

**Pull request merged**